### ISSUE TITLE
outline server installation error: "safe_docker:command not found"

### ISSUE THREAD

**Issue by mzbcracker** [Jigsaw-Code/outline-server#160]
hi.
i installed docker but it gave me the following error:
safe_docker:command not found
and also it gives me only 1 IP address for singe access key.i do not want several access keys.the single access key is enough for me.
i unblocked the only IP that it gave to me and i could connect to Server Manager using windows program.but clients cannot connect.the reason is that when i run the install scriptit gives an error and it does not give me the ip address for {ACCESS_KEY_PORT}.i know that i should openthe necessary port.but there is an error that it does not give me the port.
what seems to be the problem??

**Comment by alalamav** [comment]
Thanks for the feedback and sorry that you ran into this issue. This is being addressed in #155.

**Comment by mzbcracker** [comment]
thanks for the reply.it didn't fix the problem.still it does not give me the second port.
i also ran the installation script from alalama-safe-docker branch and still the same problem.
what should i do?

**Comment by alalamav** [comment]
Could you please paste the output of running the install script in the `alalama-safe-docker` branch? I believe the port may be a separate issue.

**Comment by mzbcracker** [comment]
here's the output:
`root@nodejs-s-1vcpu-1gb-fra1-01:~# bash -c "$(wget -qO- https://raw.githubuserco                                                                             ntent.com/Jigsaw-Code/outline-server/alalama-safe-docker/src/server_manager/inst                                                                             all_scripts/install_server.sh)"
> Verifying that Docker is installed .......... OK
> Verifying Docker permissions ................ OK
> Verifying that Docker daemon is running ..... OK
> Creating persistent state dir ............... OK
> Generating secret key ....................... OK
> Generating TLS certificate .................. OK
> Generating SHA-256 certificate fingerprint .. OK
> Starting Shadowbox .......................... OK
> Starting Watchtower ......................... OK
> Waiting for Outline server to be healthy .... OK
> Creating first user ......................... OK
> Adding API URL to config .................... OK
> Checking host firewall ...................... OK

CONGRATULATIONS! Your Outline server is up and running.

To manage your Outline server, please copy the following line (including curly
brackets) into Step 2 of the Outline Manager interface:

{"apiUrl":"https://---.---.---.---:52791/6k-B8heFr4JmXFv8AzMEdg","certSha256":"0696E809B5E71179EEBFE0BFF53A67FE120D675B1B6FBA0A76AA140EEC1156C5"}

If you have connection problems, it may be that your router or cloud provider
blocks inbound connections, even though your machine seems to allow them.

- If you plan to have a single access key to access your server make sure
  ports 52791 and  are open for TCP and UDP on
  your router or cloud provider.
- If you plan on adding additional access keys, you’ll have to open ports
  1024 through 65535 on your router or cloud provider since the Outline
  Server may allocate any of those ports to new access keys.
`
as you can see the output gives me only one port to open.
i'm really confused.
by the way i have ubuntu linux 4.13.0-43-generic

**Comment by sailosha** [comment]
you need use outline manager to connect ur server, then use manager to give other ports for the clients

**Comment by mzbcracker** [comment]
thanks for the reply.
infect the output should give us 2 ports.one for connecting using "outline manager".and another port for the "single access key" which is created by default. If we need to create additional access keys then we need to open other ports.
i do not want to open all ports from 1024 to 65535.but i think what sailosha says is right.
is there any way for us to know which port is used for an access key , when it is created in "outline manager".because without knowing that we have to open almost all ports.
is there any config file or something else to see the opened port for access keys??
i really appreciate any answer.we really need outline

**Comment by mzbcracker** [comment]
I found a config file in /shadowbox/persisted-state/ folder named "shadowbox_config.jason".
in this file i found all "access key" and their config including "access key" ports.i allowed the port and the problem solved.
i really appreciate your help.it really helped us and we really needed outline because we are heavily under Internet censorship.

**Comment by fortuna** [comment]
It's best if you use the API to get the list of access keys. See https://github.com/Jigsaw-Code/outline-server/blob/master/src/shadowbox/README.md
